docs/assignment2.md

```diff
@@ -1 +1,233 @@
+
+#### **Match**
+
+A match simulates a single soccer game between two teams.
+
+Attributes
+
+* *team1* (Type: Team): first team in the match
+* *team2* (Type: Team): second team in the match
+
+Operations
+
+- *Team getWinner()*: returns the winner of a match as described in **FR8** and **FR9**
+
+Assiciations
+
+A match is formed of exactly 2 **Team** objects.
+
+### *Views* package
+
+Each class in this package is an extention of the **GamePane** abstract class, described below. Each uses the *javafx* library and represents a different window in the GUI. All interactions with the *models* package are made via the **Game** class methods.
+
+#### **GamePane**
+
+Abstract methods
+
+- *void connectComponents()*: Connects all components into the root of a *javafx.scene.Scene*
+- *void styleComponents()*: Styles all components as required
+- *void setCallbacks()*: Sets callbacks for all interactive components
+
+### *Controllers* package
+
+Only consists of class **SceneManager** used for managing the *javafx* Scenes.
+
+Operations
+
+- *void setStage(final Stage stage)*: Sets the primary stage
+- *void showScene(final Scene scene)*: Replaces the currently active Scene with another one
+- *void showPane(final Class<? extends GamePane> pane)*: Replaces the current  **GamePane** with another
+- *SceneManager getInstance()*: returns the instance of the class
+
+
+## Object diagrams								
+Author(s): `Torstein Thomassen`
+
+This object diagram shows an example of objects with different states in our program.
+All the different attributes and variables has to be fulled in with example data.
+
+In this example each team has only three players. This is only for demonstration and will be changed for the final version.
+
+This structure resembles a tree structure with branches and leafs. The tree structure is commonly used in many software design. It's an efficient and a good way to structure the relation between objects. 
+
+This creates a hierarchy where a parent can only directly affect the children. E.g the **Match** can't affect the **Player** 'directly', only the **Team** objects.       
+
+`Object diagram over the Match Class`
+![](imgs/ass2%20obj%20diagram.png)
+
+The diagram above shows an example over the *Match* class. It has a relation with two *Team* object, and the *Team* object again has a relation with various *Player* object
+As you can see *Team 1* would win the match because they have a higher score. You can also see it's the second round, because both teams have already one win.  
+
+
+
+## State machine diagrams									
+Author(s): `Torstein Thomassen` & `Teona Moga`
+
+
+
+In the way our system is designed, only the **Game** and **League** classes are not stateless. Their state diagrams are shown below.
```

Comment:
**[SEVERE]** Player, Team are both stateful classes. 

Examples : 
- A player is either in a team or available
- A Team is either full or not full